sample-apps/go-sample-app/collection/random_metrics.go

```diff
@@ -0,0 +1,171 @@
+	rmc.registerThreadsActive()
+	rmc.registerTimeAlive()
+	rmc.registerCpuUsage()
+	return rmc
+}
+
+// registerTimeAlive registers a Synchronous Counter called TimeAlive.
+func (rmc *randomMetricCollector) registerTimeAlive() {
+	timeAlive, err := rmc.meter.SyncInt64().Counter(
+		SERVICE_NAME+"."+API_TIME_ALIVE,
+		instrument.WithDescription("Total amount of time that the application has been alive"),
+		instrument.WithUnit("ms"),
+	)
+	if err != nil {
+		fmt.Println(err)
+	}
+	rmc.timeAlive = timeAlive
+}
+
+// registerCpuUsage registers an Asynchronous Gauge called CpuUsage.
+func (rmc *randomMetricCollector) registerCpuUsage() {
+	cpuUsage, err := rmc.meter.AsyncInt64().Gauge(
+		SERVICE_NAME+"."+API_CPU_USAGE,
+		instrument.WithDescription("Cpu usage percent"),
+		instrument.WithUnit("1"),
+	)
+	if err != nil {
+		fmt.Println(err)
+	}
+	rmc.cpuUsage = cpuUsage
+
+}
+
+// registerHeapSize registers an Asynchronous UpDownCounter called HeapSize.
+func (rmc *randomMetricCollector) registerHeapSize() {
+	totalHeapSize, err := rmc.meter.AsyncInt64().UpDownCounter(
+		SERVICE_NAME+"."+API_TOTAL_HEAP_SIZE,
+		instrument.WithDescription("The current total heap size"),
+		instrument.WithUnit("By"),
+	)
+	if err != nil {
+		fmt.Println(err)
+	}
+	rmc.heapSize = totalHeapSize
+
+}
+
+// registerThreadsActive registers a Synchronous UpDownCounter called ThreadsActive.
+func (rmc *randomMetricCollector) registerThreadsActive() {
+	threadsActive, err := rmc.meter.SyncInt64().UpDownCounter(
+		SERVICE_NAME+"."+API_THREADS_ACTIVE,
+		instrument.WithUnit("1"),
+		instrument.WithDescription("The total amount of threads active"),
+	)
+	if err != nil {
+		fmt.Println(err)
+	}
+	rmc.threadsActive = threadsActive
+}
+
+// UpdateMetricsClient generates new metric values for Synchronous instruments every TimeInterval and
+// Asynchronous instruments every CollectPeriod configured by the controller.
+func (rmc *randomMetricCollector) RegisterMetricsClient(ctx context.Context, cfg Config) {
+	go func() {
+		for {
+			rmc.updateTimeAlive(ctx, cfg)
+			rmc.updateThreadsActive(ctx, cfg)
+			time.Sleep(time.Second * time.Duration(cfg.TimeInterval))
+		}
+	}()
+	rmc.updateCpuUsage(ctx, cfg)
+	rmc.updateTotalHeapSize(ctx, cfg)
+}
+
+// updateTimeAlive updates TimeAlive by TimeAliveIncrementer increments.
+func (rmc *randomMetricCollector) updateTimeAlive(ctx context.Context, cfg Config) {
+	rmc.timeAlive.Add(ctx, cfg.TimeAliveIncrementer*1000, randomMetricCommonLabels...) // in millisconds
+}
+
+// updateCpuUsage updates CpuUsage by a value between 0 and CpuUsageUpperBound every SDK call.
+func (rmc *randomMetricCollector) updateCpuUsage(ctx context.Context, cfg Config) {
+
+	if err := rmc.meter.RegisterCallback(
+		[]instrument.Asynchronous{
+			rmc.cpuUsage,
+		},
+		// SDK periodically calls this function to collect data.
+		func(ctx context.Context) {
+			min := 0
+			max := int(cfg.CpuUsageUpperBound)
+			cpuUsage := int64(rand.Intn(max-min) + min)
```

Comment:
```suggestion
	min := 0
	max := int(cfg.CpuUsageUpperBound)
	if err := rmc.meter.RegisterCallback(
		[]instrument.Asynchronous{
			rmc.cpuUsage,
		},
		// SDK periodically calls this function to collect data.
		func(ctx context.Context) {
			cpuUsage := int64(rand.Intn(max-min) + min)
```

`max` and `min` don't change over the life of this callback, so they can be defined once and closed over rather than being redefined in each invocation.

Reply:
Updated. Removed min and max from callback